[DX-955] Add use-case based SDK references across Pub/Sub pages (#3340)
ably/docs: src/pages/docs/messages/annotations.mdx

Message annotations are currently in Public Preview. That means that we are committed to supporting it and expect the API to be stable unless we discover a significant issue, but it may not yet be implemented in all of our SDKs, and there may be other minor shortcomings.
</Aside>

> <a id="interactions"/>

Reviewer: What's this?

Message annotations are currently in Public Preview. That means that we are committed to supporting it and expect the API to be stable unless we discover a significant issue, but it may not yet be implemented in all of our SDKs, and there may be other minor shortcomings.
</Aside>

<Aside data-type='note'>

> For chat applications, the [Chat SDK](/docs/chat) provides dedicated APIs for [message reactions](/docs/chat/rooms/message-reactions), [read receipts](/docs/chat/rooms/read-receipts), and [moderation](/docs/chat/moderation). Message annotations are intended for adding metadata to Pub/Sub messages outside of chat contexts. See [product guidance](/docs/platform/products) for a full comparison.

Reviewer 2: Trim "intended for adding metadata to Pub/Sub messages outside of chat contexts" — `outside of chat contexts` reads oddly. Active form is shorter:

```suggestion
For chat applications, the [Chat SDK](/docs/chat) provides dedicated APIs for [message reactions](/docs/chat/rooms/message-reactions), [read receipts](/docs/chat/rooms/read-receipts), and [moderation](/docs/chat/moderation). Use message annotations to attach metadata to Pub/Sub messages outside chat applications. See [product guidance](/docs/platform/products) for a full comparison.
```

Author: Keeping the original wording. "Message annotations are intended for adding metadata to Pub/Sub messages outside of chat contexts" frames the use case more explicitly than the imperative "Use message annotations to attach metadata...". The descriptive form makes it clearer to readers comparing options when annotations are the right tool.

Author: I'm not in favor using `attach`, since for `pub-sub` we frequently use `attach` to explicitly `attach` the channel, there shouldn't be confusion between the two.

Reviewer: As above: I don't think we should be trying to 'sell' Chat on this page. If you're looking at Chat docs then you have an annotations section there.

Author: https://github.com/ably/docs/pull/3340/changes/fe2d1956260957f50ade0818f3ea54c62f772eb0

</Aside>

<a id="interactions"/>

Message annotations enable clients to append information to existing messages on a channel. You can use annotations to implement features like:

> * **Message reactions** - add emoji reactions (👍, ❤️, 😂) to messages (for chat applications, see [Chat SDK message reactions](/docs/chat/rooms/message-reactions))

Reviewer 2: The Aside above already steers chat readers to the Chat SDK for reactions, so this trailing parenthetical duplicates the point. Drop it:

```suggestion
* **Message reactions** - add emoji reactions (👍, ❤️, 😂) to messages
```

Reviewer 2: I checked the page visually and I agree with this assessment from Claudius

Author: Applied — staged in working tree.

Author: Fixed https://github.com/ably/docs/pull/3340/commits/1a27dd4c14df6ad2f23a2ba07baaeb1b99a79004
